FlowCrypt/Functionality/DataManager/SessionManager.swift

```diff
@@ -48,6 +49,20 @@ final class SessionManager {
 
     private lazy var logger = Logger.nested(Self.self)
 
+    var currentSession: SessionType? {
+        get throws {
```

Comment:
What about setting `currentSession` on `SessionManager` initialization (as we already pass `encryptedStorage` there)?
So we don't need to add `try/catch` for each `currentSession` getter

Reply:
Sounds good. Just changed 👍

Reply:
@sosnovsky I had to revert changes because if we don't use getter, app won't get current `currentSession` value.
Like when user clicks `sign in with google` button & after auth confirm, it doesn't get session for current logged in user.

Reply:
@ioanmo226 probably this can be fixed by updating `currentSession` property in `startSessionFor(session: SessionType)`, can you try it?

Reply:
You are right. 👍 
That way works fine.